Jenkinsfile

```diff
@@ -0,0 +1,63 @@
+#!/bin/env groovy
+
+node('us-east-1 && ubuntu && docker && !gpu') {
+
+    stage('Checkout') {
+        checkout scm
+    }
+    sh "`aws ecr get-login --region=us-east-1 --no-include-email`"
+    def dockerfileChecksum = sh(returnStdout: true, script: 'md5sum Dockerfile | cut -d" " -f1').trim()
+    def dockerTag = "${dockerfileChecksum}"
+    def baseImageName = "browser-f/android:${dockerTag}"
+    stage('Build docker image') {
+        docker.withRegistry('https://141047255820.dkr.ecr.us-east-1.amazonaws.com'){
+                try {
+                    def image = docker.image(baseImageName)
+                    image.pull()
+                } catch (e) {
+                    print e
+                    def baseImage  = docker.build(baseImageName, '--build-arg UID=`id -u` --build-arg GID=`id -g` .')
+                    baseImage.push dockerTag
+                }
+        }
+    }
+
+    def args = "-u 0:0 -v ${pwd}/artifacts:/artifacts:rw"
+    docker.image("141047255820.dkr.ecr.us-east-1.amazonaws.com/${baseImageName}").inside(args) {
+        stage('Rust Setup') {
+            sh '''#!/bin/bash -l
+                curl https://sh.rustup.rs -sSf | sh -s -- -y -v
+            '''
+        }
+        stage('MozConfig'){
+            sh'''#!/bin/bash -l
+                set -x
+                set -e
+                export SHELL=/bin/bash
+                cd mozilla-release
+                rm -f mozconfig
+                echo "# Build Firefox for Android:
+                ac_add_options --enable-application=mobile/android
+                ac_add_options --target=i386-linux-android
+                # With the following Android SDK:
+                ac_add_options --with-android-sdk=\"/root/.mozbuild/android-sdk-linux\"
+                # Enable artifact building:
+                ac_add_options --enable-artifact-builds
+                # Write build artifacts to:
+                mk_add_options MOZ_OBJDIR=./objdir-frontend
+                " >> mozconfig
+            '''
+        }
+        stage('Compile APK') {
+            sh '''#!/bin/bash -l
+                export SHELL=/bin/bash
+                cd mozilla-release
+                ./mach build
+                ./mach package
```

Comment:
please separate `package` into own stage

Reply:
@sharath-cliqz Can you please comment to the Questions by @chrmod regarding Jenkinsfile and Dockerfile